### PR TITLE
removing some defaults that should not have been default

### LINKED ISSUE #139: ServerAlias (or other attributes) shouldn't be assumed if undefined.
https://github.com/rackspace-cookbooks/pythonstack/pull/138 introduced default values we should not have.

By example if we don't define `server_alias` attribute we will end up with `test.mysite.com`. 
The correct behaviour should be to not have ServerAlias if we've not defined one.

An other example, if you've not defined [repository](https://github.com/rackspace-cookbooks/phpstack/blob/master/recipes/default_unless.rb#L32), but enabled [code_deployment](https://github.com/rackspace-cookbooks/phpstack/blob/master/recipes/application_php.rb#L88), you should have an error rather than silently installing the wrong application.

### PR DESCRIPTION
**Pull request by prometheanfire**
end users don't necessarily want to have an alias called
www.subsite.example.com or want to deploy our app by default (perhaps
thy have their own app deployment method).

fixes #139

Signed-off-by: Matthew Thode <mthode@mthode.org>

### CODE REVIEW

**Comment by prometheanfire** [comment]
test this please

**Comment by prometheanfire** [comment]
hold on this, I think we need a min of one alias, so alias.first works

File: recipes/application_python.rb [hunk 1]
```diff
@@ -95,6 +95,7 @@
 if node.deep_fetch(stackname, 'code-deployment', 'enabled')
   node[stackname][node[stackname]['webserver']]['sites'].each do |port, sites|
     sites.each do |site_name, site_opts|
+      next if site_opts['repository'] == ''
```
Comment (jujugrrr) [inline recipes/application_python.rb]: maybe we don't need to default to en empty string and test an empty string, we could just do a `if site_opts['repository']` ?

Except this at least it fixes those 2 attributes for now. We can look later if this approach default_unless approach breaks something.

Comment (prometheanfire) [inline recipes/application_python.rb]: if repository is unset that check would fail, that's why we default to an empty string

Comment (martinb3) [inline recipes/application_python.rb]: @prometheanfire I'm confused... if `site_opts.key?('repository')` is false (if it's unset), `site_opts['repository']` still evaluates to `false` when coerced into a boolean value. It won't throw a Nil exception. Only accessing a subkey would throw an error, like `site_opts['repository']['foo']`.

Comment (prometheanfire) [inline recipes/application_python.rb]: makes sense, guess we need to see if the application cookbook allows a nil value for the git repo then

**Comment by martinb3** [comment]
LGTM. @jujugrrr?
